### PR TITLE
Refactor: replace FeatureFlag usage with SettingsManifest

### PR DESCRIPTION
**Pull request by bion**
### Description

This change removes `FeatureFlags` and `FeatureFlag` in favor of using `SettingsManifest`. It does not make behavioral changes to the app.

This is a prerequisite for making feature flags admin-writeable.

### Issue(s) this completes

Fixes https://github.com/civiform/civiform/issues/4948

### CODE REVIEW

File: server/app/services/settings/AbstractSettingsManifest.java [hunk 1]
```diff
@@ -4,21 +4,66 @@
...
+    for (SettingDescription settingDescription : getAllFeatureFlags()) {
+      String name = settingDescription.variableName();
+      map.put(name, getBool(name, request));
+    }
+
+    return map.build();
+  }
+
+  private ImmutableList<SettingDescription> getAllFeatureFlags() {
+    if (!getSections().containsKey("Feature Flags")) {
```
Comment (dkatzz) [inline server/app/services/settings/AbstractSettingsManifest.java:44]: **nitpick:** (Optional) You could extract "Feature Flags" to a variable FEATURE_FLAGS_SECTION_NAME or something.

File: server/app/services/settings/SettingsManifest.java [hunk 2]
```diff
@@ -143,10 +144,18 @@ public Optional<String> getLoginRadiusPrivateKeyPass() {
    * Enables [central
    * logout](https://docs.civiform.us/contributor-guide/developer-guide/authentication-providers#logout).
    */
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/services/settings/SettingsManifest.java:146]: No, this is just to support https://staging-aws.civiform.dev/dev/feature

File: server/app/services/settings/AbstractSettingsManifest.java (hunk 1 above)
Comment (bion) [reply server/app/services/settings/AbstractSettingsManifest.java:44]: Done

File: server/app/services/settings/SettingsManifest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkatzz) [reply server/app/services/settings/SettingsManifest.java:146]: Oh yes thanks!

File: server/app/services/settings/SettingsManifest.java [hunk 3]
```diff
@@ -639,77 +686,149 @@ public Optional<Integer> getDurableJobsThreadPoolSize() {
...
-  public Optional<Boolean> getFeatureFlagOverridesEnabled() {
+  public boolean getFeatureFlagOverridesEnabled() {
     return getBool("FEATURE_FLAG_OVERRIDES_ENABLED");
   }
 
+  /**
+   * Allows feature flags to be overridden via request cookies. Is used by browser tests. Should
+   * only be enabled in test and staging deployments.
+   */
+  public boolean getFeatureFlagOverridesEnabled(Request request) {
```
[the remark this answers is not in the document]
Comment (dkatzz) [reply server/app/services/settings/SettingsManifest.java:752]: But getBool calls `overridesEnabled`, which calls `getBool("FEATURE_FLAG_OVERRIDES_ENABLED");`, not this function with the request passed through? Unless I'm looking at the wrong thing?

[the remark this answers is not in the document]
Comment (bion) [reply server/app/services/settings/SettingsManifest.java:752]: Ah, I was confused by looking at the wrong file when I initially responded. This method is generated due to `FEATURE_FLAG_OVERRIDES_ENABLED` being a flag in env-var-docs.json. The one in the AbstractSettingsManifest is needed by methods that are also defined in that class.

[the remark this answers is not in the document]
Comment (dkatzz) [reply server/app/services/settings/SettingsManifest.java:752]: Oh got it! Thank you!!

**Review by dkatzz — APPROVED** [review]

**Comment by dkatzz** [comment]
Just a reminder to update the docs as well :)

File: server/app/services/settings/AbstractSettingsManifest.java [hunk 4]
```diff
@@ -4,21 +4,67 @@
...
+
+    for (SettingDescription settingDescription : getAllFeatureFlags()) {
+      String name = settingDescription.variableName();
+      map.put(name, getBool(name, request));
+    }
+
+    return map.build();
+  }
+
+  private ImmutableList<SettingDescription> getAllFeatureFlags() {
```
Comment (rockycodes) [inline server/app/services/settings/AbstractSettingsManifest.java:44]: suggestion (non-blocking): rename this function `getAllFeatureFlagsSettingDescriptions` to make it more clear what the function is

File: server/app/services/settings/AbstractSettingsManifest.java [hunk 5]
```diff
@@ -4,21 +4,67 @@
...
 import java.util.Locale;
 import java.util.Optional;
 import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.mvc.Http;
 
 /** Provides behavior for {@link SettingsManifest}. */
 public abstract class AbstractSettingsManifest {
+  private static final String FEATURE_FLAG_SETTING_SECTION_NAME = "Feature Flags";
```
Comment (rockycodes) [inline server/app/services/settings/AbstractSettingsManifest.java:19]: suggestion (non-blocking): I like making this a const. I suggest using it other places we're currently using the string "Feature Flags" if it makes sense to. Specifically the `GENERATED_SECTIONS` var in `SettingsManifest.java` to provide extra assurance that that section will be found.

**Review by rockycodes — APPROVED** [review]
what an update! overall looks good, just a couple of non-blocking comments.

File: server/app/services/settings/AbstractSettingsManifest.java (hunk 4 above)
Comment (bion) [reply server/app/services/settings/AbstractSettingsManifest.java:44]: Done

File: server/app/services/settings/AbstractSettingsManifest.java (hunk 5 above)
Comment (bion) [reply server/app/services/settings/AbstractSettingsManifest.java:19]: That's a bit tricky given SettingsManifest.java is generated based on the contents of env-var-docs.json, where `"Feature Flags"` is one of the section titles

**Comment by rockycodes** [comment]
I just ran this locally and realized the the feature flags that are showing up in the dev page are different now. Do you know why that might be?

without these changes:
image

with these changes:
image

File: server/app/services/settings/AbstractSettingsManifest.java (hunk 5 above)
Comment (rockycodes) [reply server/app/services/settings/AbstractSettingsManifest.java]: Ah gotcha, never mind then.

**Comment by bion** [comment]
@rockycodes great catch! I added the ESRI flags to the feature flags section of env-var-docs.json so they show up on the overrides page now

**Pull request merged**